services/navigation/builtin/builtin.go

```diff
@@ -384,10 +408,17 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 				extra,
 			)
 			if err != nil {
+				svc.logger.Infof("hit motion error %s when navigating to waypoint %+v", err.Error(), wp)
 				return err
 			}
 
-			return svc.waypointReached(ctx)
+			if !success {
+				err := errors.New("failed to reach goal")
+				svc.logger.Infof("hit motion failed to reach goal %s when navigating to waypoint %+v", err.Error(), wp)
```

Comment:
Since we always define "err" to be "failed to reach goal", this will always double up that string.

Reply:
good catch, fixed